src/utils/registry.ts

```diff
@@ -232,123 +207,184 @@ function readDescriptors(packagePath: string) {
   }
 
-  async validateHostRequirements(browserName: BrowserName) {
+  private async _validateHostRequirements(browserName: BrowserName, browserDirectory: string, linuxLddDirectories: string[], dlOpenLibraries: string[], windowsExeAndDllDirectories: string[]) {
     if (getAsBooleanFromENV('PLAYWRIGHT_SKIP_VALIDATE_HOST_REQUIREMENTS')) {
       process.stdout.write('Skipping host requirements validation logic because `PLAYWRIGHT_SKIP_VALIDATE_HOST_REQUIREMENTS` env variable is set.\n');
       return;
     }
     const ubuntuVersion = await getUbuntuVersion();
-    if ((browserName === 'firefox' || browserName === 'firefox-beta') && ubuntuVersion === '16.04')
-      throw new Error(`Cannot launch ${browserName} on Ubuntu 16.04! Minimum required Ubuntu version for Firefox browser is 18.04`);
-    const browserDirectory = this.browserDirectory(browserName);
-
-    if (os.platform() === 'linux') {
-      const dlOpenLibraries: string[] = [];
-      const linuxLddDirectories: string[] = [];
-      if (browserName === 'chromium' || browserName === 'chromium-with-symbols')
-        linuxLddDirectories.push(path.join(browserDirectory, 'chrome-linux'));
-      if (browserName === 'webkit') {
-        linuxLddDirectories.push(
-            path.join(browserDirectory, 'minibrowser-gtk'),
-            path.join(browserDirectory, 'minibrowser-gtk', 'bin'),
-            path.join(browserDirectory, 'minibrowser-gtk', 'lib'),
-            path.join(browserDirectory, 'minibrowser-wpe'),
-            path.join(browserDirectory, 'minibrowser-wpe', 'bin'),
-            path.join(browserDirectory, 'minibrowser-wpe', 'lib'),
-        );
-        dlOpenLibraries.push('libGLESv2.so.2', 'libx264.so');
-      }
-      if (browserName === 'firefox' || browserName === 'firefox-beta')
-        linuxLddDirectories.push(path.join(browserDirectory, 'firefox'));
-      return await validateDependenciesLinux(linuxLddDirectories, dlOpenLibraries);
-    }
+    if (browserName === 'firefox' && ubuntuVersion === '16.04')
+      throw new Error(`Cannot launch Firefox on Ubuntu 16.04! Minimum required Ubuntu version for Firefox browser is 18.04`);
 
-    if (os.platform() === 'win32' && os.arch() === 'x64') {
-      const windowsExeAndDllDirectories: string[] = [];
-      if (browserName === 'chromium' || browserName === 'chromium-with-symbols')
-        windowsExeAndDllDirectories.push(path.join(browserDirectory, 'chrome-win'));
-      if (browserName === 'firefox' || browserName === 'firefox-beta')
-        windowsExeAndDllDirectories.push(path.join(browserDirectory, 'firefox'));
-      if (browserName === 'webkit')
-        windowsExeAndDllDirectories.push(browserDirectory);
-      return await validateDependenciesWindows(windowsExeAndDllDirectories);
-    }
+    if (os.platform() === 'linux')
+      return await validateDependenciesLinux(linuxLddDirectories.map(d => path.join(browserDirectory, d)), dlOpenLibraries);
+    if (os.platform() === 'win32' && os.arch() === 'x64')
```

Comment:
Also not fully related to the PR, but this makes me feel bad bad when I see it - tried to rework it locally but was not happy with the approach.

Problem: When a user has installed the 32 bit version of Node.js on a 64 bit Windows system which quite a few user by miss-installation do, os.arch() returns `ia32` which has some major disadvantages.
- Wrong statistical metrics -> we get too much downloads for ia32 and think these are actually real ia32 systems (main goal: drop unused architectures but we can't due wrong downloads get generated)
- Performance downsides -> 32bit browsers get used under 64 bit windows
- When the user has installed a different language binding, that finds out the system arch correctly like Python to download chromium -> it downloads 64bit ffmpeg, then uses wrongly installed Node.js (32 bit) -> it tries to download 32 bit ffmpeg - (os arch based executable path) - but sees the folder which is there and thinks it there. Still internally when launching ffmpeg it fails, see #6946

Action item at some point: We should change how we determine if its a 32 or 64 bit system on Windows, maybe through the `PROCESSOR_ARCHITEW6432 ` env var, but unsure. It needs to be the same on all the language bindings.